lib/internal/util.js

```diff
@@ -532,7 +533,10 @@ function defineLazyProperties(target, id, keys, enumerable = true) {
       mod ??= require(id);
       if (lazyLoadedValue === undefined) {
         lazyLoadedValue = mod[key];
-        set(lazyLoadedValue);
+        if (!ObjectIsFrozen(this)) {
```

Comment:
I don't think `isFrozen` is the correct check for that, I think `isSealed` would be closer to what we're after, but really what we want to know is wether the property is configurable.

Reply:
Good catch, fixed and added another test for Object.seal(). Thanks!